src/lasp_core.erl

```diff
@@ -912,6 +912,34 @@ write(Type, Value, Key, Store) ->
 increment_counter(Counter) ->
     Counter + 1.
 
+%% @private
+store_delta(Type, Counter, Delta, DeltaMap0) ->
+    MaxDeltaSlots = application:get_env(lasp, delta_mode_max_slots, 10),
+    %% Check the space of the DeltaMap
+    case orddict:size(DeltaMap0) < MaxDeltaSlots of
+        true ->
+            %% Store a new delta.
+            orddict:store(Counter, Delta, DeltaMap0);
+        false ->
+            DeltaCounterList = orddict:fetch_keys(DeltaMap0),
+            %% Find the minimum counter & its delta.
+            MinCounter0 = lists:nth(1, DeltaCounterList),
+            MinCounterDelta0 = orddict:fetch(MinCounter0, DeltaMap0),
+            %% Find the 2nd minimum counter & its delta.
+            MinCounter1 = lists:nth(2, DeltaCounterList),
+            MinCounterDelta1 = orddict:fetch(MinCounter1, DeltaMap0),
```

Comment:
This section here, from L924-930 is traversing the list multiple times and is very expensive.  (Once to grab all keys, again to get the first element, then again to get it from the dict, then again to get the second element, then again to get the delta from the dict.)  Can we just convert it to a list once with `to_list` at the beginning of the execution and operate directly on it as a list?  

When you want to grab the first two minimal counters, you could easily do [First, Second | _Rest] and save a bit of computation that you have in your example above.